### PR TITLE
Improve code readability in dict.c

### PR DESCRIPTION
**Pull request by PingXie**
This pull request improves code readability, as a follow up of #749.

- Internal Naming Conventions: Removed the use of underscores (_) for internal static structures/functions.

- Descriptive Function Names: Updated function names to be more descriptive, making their purpose clearer. For instance, `_dictExpand` is renamed to `dictExpandIfAutoResizeAllowed`.

### CODE REVIEW

File: src/dict.c [hunk 1]
```diff
@@ -91,21 +91,21 @@ typedef struct {
     struct dictEntry *next;  /* Next entry in the same hash bucket. */
     uint8_t key_header_size; /* offset into key_buf where the key is located at. */
     unsigned char key_buf[]; /* buffer with embedded key. */
-} embeddedDictEntry;
+} dictEntryEmbedded;
 
-/* Validation and helper for `embeddedDictEntry` */
+/* Validation and helper for `_dictEntryEmbedded` */
```
Comment (enjoy-binbin) [inline src/dict.c]:
```suggestion
/* Validation and helper for `dictEntryEmbedded` */
```

**Review by enjoy-binbin — APPROVED** [review]

**Comment by madolson** [comment]
@PingXie Ping that you need to rebase this.

**Comment by hpatro** [comment]
> @PingXie Ping that you need to rebase this.

yeah, on first look I thought this change was merged :D

**Comment by PingXie** [comment]
I almost forgot about this too. Thanks for the reminder!

**Comment by PingXie** [comment]
@madolson @enjoy-binbin ready for your review

**Comment by enjoy-binbin** [comment]
The unit test failed: _dictNextExp

https://github.com/valkey-io/valkey/actions/runs/10844745430/job/30094212586?pr=943#step:3:682

**Review by madolson — DISMISSED** [review]

**Review by hwware — APPROVED** [review]

**Comment by hwware** [comment]
Thanks for renaming the function and following C traditional.

**Pull request merged**